### ISSUE TITLE
I have meet this problem in my cGAN project:No gradients provided for any variable, check your graph for ops that do not support gradients, What should I do to?should I change my loss function?

### ISSUE THREAD

**Issue by TimmyFan** [TimmyFan/RNA-second-structure#1]
import glob
import os
import random
from ops import *
matrix_grey_dir = r'C:/Users/49610/.spyder-py3/tRNA_location_matrix/'
matrix_grey_paths = glob.glob(matrix_grey_dir + '*.csv')

matrix_RGB_dir = r'C:/Users/49610/.spyder-py3/tRNA_bptype_matrix/'
matrix_RGB_paths = glob.glob(matrix_RGB_dir + '*.csv')

matrix_label_dir = r'C:/Users/49610/.spyder-py3/tRNA_label_bool/'
matrix_label_paths = glob.glob(matrix_RGB_dir + '*.csv')

global sample
for i in range(706):
    if i<1:
        sample_grey = np.loadtxt(open(matrix_grey_paths[i], "rb"), delimiter=",")
        #sample_RGB = np.loadtxt(open(matrix_RGB_paths[i], "rb"), delimiter=",")
        #sample = np.array([sample_grey, sample_RGB])
        sample = np.array([sample_grey])
    if i>=1 :
        sample_grey = np.loadtxt(open(matrix_grey_paths[i], "rb"), delimiter=",")
        #sample_RGB = np.loadtxt(open(matrix_RGB_paths[i], "rb"), delimiter=",")
        sample1 = np.array([sample_grey])
        sample = np.concatenate((sample, sample1), axis=1)
sample = tf.transpose(sample)
sample = tf.reshape(sample, [-1,400,400] )

global label
for i in range(706):
    if i<1:
        label = np.loadtxt(open(matrix_label_paths[i], "rb"), delimiter=",")
        # sample = tf.convert_to_tensor(sample)
    if i>=1:
        label1 = np.loadtxt(open(matrix_label_paths[i], "rb"), delimiter=",")
        label = np.concatenate((label, label1), axis=1)
label = tf.reshape(label, [-1,400,400] )
global_step = tf.Variable(0, name='global_step', trainable=False)
tf.reset_default_graph()
num = random.randint(0,705)
num1 = num-20
train_feed_feature = sample[num1:num,0:400,0:400]
train_feed_label = label[num1:num, 0:400,0:400]
y = tf.placeholder(tf.float32, [BATCH_SIZE, 400, 400], name='y')
G_result = tf.placeholder(tf.float32, [BATCH_SIZE, 400, 400], name='g_result')
images = tf.placeholder(tf.float32, [BATCH_SIZE, 400, 400], name='real_images')

# G是生成的假图片
with tf.variable_scope(tf.get_variable_scope()) as scope:# D、D_lo
    G = generator(y)
    D_real, D_logits_real = discriminator(images)  #gits都是 BATCH_SIZE*1的
    D_fake, D_logits_fake = discriminator1(G_result,reuse = True)

# 损失计算

d_loss_real = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(logits=D_logits_fake, labels= D_real))
d_loss_fake = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(logits=D_logits_real, labels= D_real))
D_loss = d_loss_fake+d_loss_real
G_loss = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(logits=D_logits_fake, labels= D_real))

# 生成器和判别器要更新的变量，用于 tf.train.Optimizer 的 var_list
t_vars = tf.trainable_variables()
d_vars = [var for var in t_vars if 'd_' or 'd1_' in var.name]
g_vars = [var for var in t_vars if 'g_' in var.name]

d_optim = tf.train.AdamOptimizer(0.0002, beta1=0.5).minimize(D_loss, var_list=d_vars, global_step=global_step)
g_optim = tf.train.AdamOptimizer(0.0002, beta1=0.5).minimize(G_loss, var_list=g_vars, global_step=global_step)

with tf.Session() as sess:
    saver = tf.train.Saver()
    sess.run(tf.global_variables_initializer())
    for epoch in range(10000):
            sess.run([d_optim], feed_dict={G_result: G, images: train_feed_label})
            sess.run([g_optim], feed_dict={y: train_feed_feature})
            if epoch % 100 == 0:
                errD = D_loss.eval(feed_dict={images: G, y: train_feed_label})
                errG = G_loss.eval({y: train_feed_feature})
                print("epoch:[%d], i:[%d]  d_loss: %.8f, g_loss: %.8f" % (epoch, i, errD, errG))
